lib/rspec/matchers/built_in/change.rb

```diff
@@ -387,8 +387,10 @@ def changed?
           #
           # Note that it is not sufficient to only check the hashes; it is
           # possible for two values to be unequal (and of different classes)
-          # but to return the same hash value.
-          @actual_before != @actual_after || @before_hash != @actual_after.hash
+          # but to return the same hash value. Also, some objects may change
+          # their hash after being compared with `==`/`!=`.
+          actual_hash = @actual_after.hash
```

Comment:
As indicated in my last comment on the old code, I'd prefer this be moved up to just after `@actual_after = evaluate_value_proc` and be an `@actual_hash` definition.

Reply:
Ah, sorry, I initially got your idea wrong. Fixed.
Does it look up to standards now?